x/gov/src/msg/proposal.rs

```diff
@@ -0,0 +1,119 @@
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct MsgSubmitProposal {
+    pub content: Any,
+    pub initial_deposit: SendCoins,
+    pub proposer: AccAddress,
+}
+
+impl MsgSubmitProposal {
+    pub const TYPE_URL: &'static str = "/cosmos.gov.v1beta1/MsgSubmitProposal";
+}
+
+impl TxMessage for MsgSubmitProposal {
+    fn get_signers(&self) -> Vec<&AccAddress> {
+        vec![&self.proposer]
+    }
+
+    fn validate_basic(&self) -> Result<(), String> {
+        Ok(())
+    }
+
+    fn type_url(&self) -> &'static str {
+        MsgSubmitProposal::TYPE_URL
+    }
+}
+
+impl Protobuf<inner::MsgSubmitProposal> for MsgSubmitProposal {}
+
+impl TryFrom<inner::MsgSubmitProposal> for MsgSubmitProposal {
+    type Error = CoreError;
+
+    fn try_from(
+        inner::MsgSubmitProposal {
+            content,
+            initial_deposit,
+            proposer,
+        }: inner::MsgSubmitProposal,
+    ) -> Result<Self, Self::Error> {
+        Ok(Self {
+            content: content.ok_or(CoreError::MissingField(
+                "MsgSubmitProposal missing content".to_owned(),
+            ))?,
+            initial_deposit: SendCoins::new({
+                let mut coins = Vec::with_capacity(initial_deposit.len());
+                for coin in initial_deposit {
+                    coins.push(
+                        coin.try_into()
+                            .map_err(|e: CoinsError| CoreError::Coin(e.to_string()))?,
+                    )
+                }
+
+                coins
+            })
+            .map_err(|e| CoreError::Coins(e.to_string()))?,
+            proposer: AccAddress::from_bech32(&proposer)
+                .map_err(|e| CoreError::DecodeAddress(e.to_string()))?,
+        })
+    }
+}
+
+impl From<MsgSubmitProposal> for inner::MsgSubmitProposal {
+    fn from(
+        MsgSubmitProposal {
+            content,
+            initial_deposit,
+            proposer,
+        }: MsgSubmitProposal,
+    ) -> Self {
+        Self {
+            content: Some(content),
+            initial_deposit: initial_deposit
+                .into_inner()
+                .into_iter()
+                .map(|e| e.into())
+                .collect(),
+            proposer: proposer.to_string(),
+        }
+    }
+}
+
+impl TryFrom<Any> for MsgSubmitProposal {
+    type Error = CoreError;
+
+    fn try_from(value: Any) -> Result<Self, Self::Error> {
+        if value.type_url != Self::TYPE_URL {
+            Err(CoreError::DecodeGeneral(
+                "message type not recognized".into(),
+            ))?
+        }
+        Ok(MsgSubmitProposal::decode::<Bytes>(value.value.into())
+            .map_err(|e| CoreError::DecodeProtobuf(e.to_string()))?)
+    }
+}
+
+impl From<MsgSubmitProposal> for Any {
+    fn from(msg: MsgSubmitProposal) -> Self {
+        Any {
+            type_url: MsgSubmitProposal::TYPE_URL.to_string(),
+            value: msg.encode_vec().expect(IBC_ENCODE_UNWRAP),
```

Comment:
If I'm correct, the newer protobuf converts  bytes without result.

Reply:
I use older from gears here.